src/Stack/Script.hs

```diff
@@ -205,20 +208,34 @@ blacklist = Set.fromList
     , mkPackageName "cryptohash-sha256"
     ]
 
-toModuleInfo :: LoadedSnapshot -> ModuleInfo
-toModuleInfo ls =
-      mconcat
-    $ map (\(pn, lpi) ->
-            ModuleInfo
-            $ Map.fromList
-            $ map (, Set.singleton pn)
-            $ Set.toList
-            $ lpiExposedModules lpi)
-    $ filter (\(pn, lpi) ->
-            not (lpiHide lpi) &&
-            pn `Set.notMember` blacklist)
-    $ Map.toList
-    $ Map.union (void <$> lsPackages ls) (void <$> lsGlobals ls)
+getModuleInfo :: HasEnvConfig env => RIO env ModuleInfo
+getModuleInfo = do
+    sourceMap <- view $ envConfigL . to envConfigSourceMap
+    installMap <- toInstallMap sourceMap
+    (_installedMap, globalDumpPkgs, snapshotDumpPkgs, _localDumpPkgs) <-
+        getInstalled
+            GetInstalledOpts
+            { getInstalledProfiling = False
+            , getInstalledHaddock = False
+            , getInstalledSymbols = False
+            }
+            installMap
+    return $
+        toModuleInfo (notHidden $ smDeps sourceMap) snapshotDumpPkgs <>
+        toModuleInfo (smGlobal sourceMap) globalDumpPkgs
```

Comment:
Same comment as below applies here when `mappend` is used. Can we have clashing module names between packages in `snapshotDumpPkgs` and `globalDumpPkgs`?

Reply:
```
instance Semigroup ModuleInfo where
  ModuleInfo x <> ModuleInfo y =
    ModuleInfo (Map.unionWith Set.union x y)
```